cpp/src/decisiontree/batched-levelalgo/builder_base.cuh

```diff
@@ -394,6 +394,7 @@ struct ClsTraits {
     dim3 grid(b.n_blks_for_rows, colBlks, batchSize);
     size_t smemSize = sizeof(int) * binSize + sizeof(DataT) * nbins;
     smemSize += sizeof(int);
+    smemSize += 3 * sizeof(DataT*);  // Room for alignment in worst case
```

Comment:
instead of this, it is better to do smem size computation as:
```
constexpr size_t kMaxSize = std::max(sizeof(int), sizeof(DataT));
size_t smemSize = alignTo(sizeof(int) * binSize, kMaxSize);
smemSize += alignTo(sizeof(DataT) * nbins, kMaxSize);
smemSize += alignTo(sizeof(int), kMaxSize);
```

Reply:
We will also need to cover the MAE-only alignments for another 4 items here... I think this is getting to be a lot of code duplication for memory size calculation, and I'd be curious about whether we can just live with an overestimate